pkg/sync/kubernetes/featureflagconfiguration/clientset.go

```diff
@@ -0,0 +1,141 @@
+package featureflagconfiguration
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"time"
+
+	"github.com/open-feature/flagd/pkg/sync"
+	ffv1alpha1 "github.com/open-feature/open-feature-operator/apis/core/v1alpha1"
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type FFCInterface interface {
+	FeatureFlagConfigurations(namespace string) Interface
+}
+
+type FFCClient struct {
+	restClient rest.Interface
+}
+
+func createFuncHandler(obj interface{}, object client.ObjectKey, c chan<- sync.INotify) error {
+	if reflect.TypeOf(obj) != reflect.TypeOf(&ffv1alpha1.FeatureFlagConfiguration{}) {
+		return errors.New("object is not a FeatureFlagConfiguration")
+	}
+	if obj.(*ffv1alpha1.FeatureFlagConfiguration).Name == object.Name {
+		c <- &sync.Notifier{
+			Event: sync.Event[sync.DefaultEventType]{
+				EventType: sync.DefaultEventTypeCreate,
+			},
+		}
+	}
+	return nil
+}
+
+func updateFuncHandler(oldObj interface{}, newObj interface{}, object client.ObjectKey, c chan<- sync.INotify) error {
+	if reflect.TypeOf(oldObj) != reflect.TypeOf(&ffv1alpha1.FeatureFlagConfiguration{}) {
+		return errors.New("old object is not a FeatureFlagConfiguration")
+	}
+	if reflect.TypeOf(newObj) != reflect.TypeOf(&ffv1alpha1.FeatureFlagConfiguration{}) {
+		return errors.New("new object is not a FeatureFlagConfiguration")
+	}
+	if oldObj.(*ffv1alpha1.FeatureFlagConfiguration).Name == object.Name {
+		// Only update if there is an actual featureFlagSpec change
+		c <- &sync.Notifier{
+			Event: sync.Event[sync.DefaultEventType]{
+				EventType: sync.DefaultEventTypeModify,
+			},
+		}
+	}
+	return nil
+}
+
+func deleteFuncHandler(obj interface{}, object client.ObjectKey, c chan<- sync.INotify) error {
+	if reflect.TypeOf(obj) != reflect.TypeOf(&ffv1alpha1.FeatureFlagConfiguration{}) {
+		return errors.New("object is not a FeatureFlagConfiguration")
+	}
+	if obj.(*ffv1alpha1.FeatureFlagConfiguration).Name == object.Name {
+		c <- &sync.Notifier{
+			Event: sync.Event[sync.DefaultEventType]{
+				EventType: sync.DefaultEventTypeDelete,
+			},
+		}
+	}
+	return nil
+}
+
+func WatchResources(l log.Entry, clientSet FFCInterface, refreshTime time.Duration,
```

Comment:
small change, but `ctx` is usually defined as the first argument of a function signature, I can't find an explicit definition for this but its mentioned [in here ](https://go.dev/blog/context-and-structs)